### PR TITLE
[building] Add build instructions for CMake

### PR DESCRIPTION
**Pull request by Fohlen**
This add's `CMake` build instructions as suggested in #14

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+## CMake
+
+To build `libsigc++` with `CMake` proceed as follows:
+
+- `cd libsigcplusplus/build`
+- `cmake ..`
+- `make`
+
+This will build the library including all it's test cases. If you wish to use different options consider `make help`
```
Comment (murraycu) [inline BUILDING.md]: its, not it's

File: BUILDING.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Building
+--------
+
+You can choose to build this library either with `cmake` or `automake`.
+The pre-requisite for compilations are:
+
+- you should have cloned this repository with `git clone https://github.com/libsigcplusplus/libsigcplusplus.git`
```
Comment (murraycu) [inline BUILDING.md]: Most people would get via a release tarball, not via git. The build is generally more difficult directly from git.

File: BUILDING.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+- you should have a modern version of `cmake` or `automake` depending on your preference
```
Comment (murraycu) [inline BUILDING.md]: This is too vague. I would just leave this and the GCC or Clang mention out. The build script will complain if it doesn't have what it needs.

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- either GCC or Clang should be installed with a version that supports at least `C++17`
+
+## CMake
+
+To build `libsigc++` with `CMake` proceed as follows:
+
+- `cd libsigcplusplus/build`
```
Comment (murraycu) [inline BUILDING.md]: Don't you have to mkdir build first?

File: BUILDING.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- `cmake ..`
+- `make`
+
+This will build the library including all it's test cases. If you wish to use different options consider `make help`
+
+## automake
```
Comment (murraycu) [inline BUILDING.md]: It's more autotools than just automake. Also, this is still the primary way to build, so it should be first, please.

File: BUILDING.md (hunk 2 above)
Comment (Fohlen) [reply BUILDING.md]: @murraycu this is generally true but it is good for people who would like to embed `libsigc++` into their software (or for package maintainers) to have build instructions ;)

File: BUILDING.md (hunk 4 above)
Comment (Fohlen) [reply BUILDING.md]: yup

File: BUILDING.md (hunk 5 above)
Comment (murraycu) [reply BUILDING.md]: ./configure and make. It's the standard way to build from source. The tarball usually contains an INSTALL file that goes into detail about that.

**Comment by Fohlen** [comment]
@murraycu thanks for checking back with me. I have fixed all the details we talked about so it can be considered ready for a second review! Happy christmas 🎄

**Comment by murraycu** [comment]
Sorry, but I added some text in the README.md about building, in response to your issue #18 before I remembered that this PR is here. I think my text is probably enough, but you might check if anything from this PR needs to be there too. Sorry.

**Pull request closed without merging**